### PR TITLE
Bump, tag, publish GHA.

### PR DESCRIPTION
**Pull request by marikomedlock**
- Added a `client` Gradle sub-project to build and publish the client library to Broad artifactory.
- Added a `publish.sh` script for manually publishing the library.
- First pass at a GHA to automatically bump the version and publish the library and service image. I don't think I can test this GHA until at least one version of it is in the main branch, so consider this a first draft and I'll tackle any problems in a follow-on PR.
- Started version at `0.0.6`.

I generally tried to follow what Terra services (wsm, rbs, user manager) are already doing, so most of this code is copied from one of those repos. One weird thing I noticed is that the client library seems to publish with a `-plain.jar` suffix. I don't know where it's getting the `plain` from, but decided to leave as is for now and come back later if this is a problem.

Not strictly part of this PR, but I also added the `BROADBOT_TOKEN` as a GH secret with this `terraform-ap-deployments` [PR](https://github.com/broadinstitute/terraform-ap-deployments/pull/856). This is needed for the bump-publish GHA.

### CODE REVIEW

File: .github/actions/bump-skip/action.yaml [hunk 1]
```diff
@@ -0,0 +1,28 @@
+# This action must be done after the checkout action
+name: 'bump-skip'
+description: 'Set skip-out when we are doing a version bump'
+author: 'dd'
```
Comment (melissachang) [inline .github/actions/bump-skip/action.yaml:4]: Delete?

File: client/gradle/open-api.gradle [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+                            "modelTests=false," +
+                            "modelDocs=false," +
+                            "dateLibrary=java8"
+            ]
+        }
+    }
+}
+
+// Internal dependencies
+//generateSwaggerCodeClient.dependsOn(':openapi:buildOpenApi')
```
Comment (melissachang) [inline client/gradle/open-api.gradle:32]: Delete lines 32, 38, 39?

**Review by melissachang — APPROVED** [review]

File: .github/actions/bump-skip/action.yaml (hunk 1 above)
Comment (marikomedlock) [reply .github/actions/bump-skip/action.yaml:4]: done.

File: client/gradle/open-api.gradle (hunk 2 above)
Comment (marikomedlock) [reply client/gradle/open-api.gradle]: done, thank you.

**Pull request merged**